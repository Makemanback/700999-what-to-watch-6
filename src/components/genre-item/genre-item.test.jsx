import React from 'react';
import {Router} from 'react-router-dom';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';

import GenreItem from './genre-item';

const itemClass = `mock class`;
const index = 1;
const changeGenreHandler = jest.fn();
const genre = `drama`;
const setActive = jest.fn();
test(`Should GenreItem render correctly`, () => {

  const history = createMemoryHistory();

  const {container} = render(
      <Router history={history}>
        <GenreItem itemClass={itemClass} index={index} genre={genre} changeGenreHandler={changeGenreHandler} setActive={setActive} />
      </Router>
  );

  expect(container).toMatchSnapshot();
});

