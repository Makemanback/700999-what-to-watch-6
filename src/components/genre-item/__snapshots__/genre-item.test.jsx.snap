// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreItem render correctly 1`] = `
<div>
  <li
    class="catalog__genres-item mock class"
  >
    <a
      class="catalog__genres-link"
      href="/"
    >
      drama
    </a>
  </li>
</div>
`;
